cli/cmd/main.go

```diff
@@ -296,6 +296,87 @@ func main() {
 				},
 				Action: DataCommand,
 			},
+			{
+				Name:  "delete",
```

Comment:
I'd lean towards calling this delete-data or deletedata but also nice to have single word commands. Getting another opinion @AaronCasas

Reply:
Hmm. I definitely think that this and the export command should be consistent in the sense that if this is `data delete`, that should be `data export/download/whatever`, and if this is just `delete`, that should just be `export/download/whatever`. I think I'd be partial towards the former (`viam data <action`) because I can definitely foresee a future where there are other delete operations we expose through the CLI (e.g. delete a robot).

Long winded way of saying: I'd vote `data delete` and changing `data` to `data export` or `data download`

Reply:
+1 to consistency and good suggestion, as solely passing `--data` isn't descriptive as we add more functionality to the CLI. I'd set this up with `data` as the main command and `export` and `delete` as the subcommands:

https://stackoverflow.com/a/54900733

Reply:
+1 to using subcommands, especially since we have so many flags here that'll all be shared

Reply:
Great! Added as a subcommand